File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-com-5-4/CoalPhysicalLeg.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class CoalPhysicalLeg : PhysicalSwapLeg
    {
        public CoalPhysicalLeg(XmlNode xmlNode)
        : base(xmlNode)
        {
            XmlNode deliveryPeriodsNode = xmlNode.SelectSingleNode("deliveryPeriods");
            
            if (deliveryPeriodsNode != null)
            {
                if (deliveryPeriodsNode.Attributes["href"] != null || deliveryPeriodsNode.Attributes["id"] != null) 
                {
                    if (deliveryPeriodsNode.Attributes["id"] != null) 
                    {
                        deliveryPeriodsIDRef_ = deliveryPeriodsNode.Attributes["id"].Value;
                        CommodityDeliveryPeriods ob = new CommodityDeliveryPeriods(deliveryPeriodsNode);
                        IDManager.SetID(deliveryPeriodsIDRef_, ob);
                    }
                    else if (deliveryPeriodsNode.Attributes["href"] != null)
                    {
                        deliveryPeriodsIDRef_ = deliveryPeriodsNode.Attributes["href"].Value;
                    }
                    else
                    {
                        deliveryPeriods_ = new CommodityDeliveryPeriods(deliveryPeriodsNode);
                    }
                }
                else
                {
                    deliveryPeriods_ = new CommodityDeliveryPeriods(deliveryPeriodsNode);
                }
            }
            
        
            XmlNode coalNode = xmlNode.SelectSingleNode("coal");
            
            if (coalNode != null)
            {
                if (coalNode.Attributes["href"] != null || coalNode.Attributes["id"] != null) 
                {
                    if (coalNode.Attributes["id"] != null) 
                    {
                        coalIDRef_ = coalNode.Attributes["id"].Value;
                        CoalProduct ob = new CoalProduct(coalNode);
                        IDManager.SetID(coalIDRef_, ob);
                    }
                    else if (coalNode.Attributes["href"] != null)
                    {
                        coalIDRef_ = coalNode.Attributes["href"].Value;
                    }
                    else
                    {
                        coal_ = new CoalProduct(coalNode);
                    }
                }
                else
                {
                    coal_ = new CoalProduct(coalNode);
                }
            }
            
        
            XmlNode deliveryConditionsNode = xmlNode.SelectSingleNode("deliveryConditions");
            
            if (deliveryConditionsNode != null)
            {
                if (deliveryConditionsNode.Attributes["href"] != null || deliveryConditionsNode.Attributes["id"] != null) 
                {
                    if (deliveryConditionsNode.Attributes["id"] != null) 
                    {
                        deliveryConditionsIDRef_ = deliveryConditionsNode.Attributes["id"].Value;
                        CoalDelivery ob = new CoalDelivery(deliveryConditionsNode);
                        IDManager.SetID(deliveryConditionsIDRef_, ob);
                    }
                    else if (deliveryConditionsNode.Attributes["href"] != null)
                    {
                        deliveryConditionsIDRef_ = deliveryConditionsNode.Attributes["href"].Value;
                    }
                    else
                    {
                        deliveryConditions_ = new CoalDelivery(deliveryConditionsNode);
                    }
                }
                else
                {
                    deliveryConditions_ = new CoalDelivery(deliveryConditionsNode);
                }
            }
            
        
            XmlNode deliveryQuantityNode = xmlNode.SelectSingleNode("deliveryQuantity");
            
            if (deliveryQuantityNode != null)
            {
                if (deliveryQuantityNode.Attributes["href"] != null || deliveryQuantityNode.Attributes["id"] != null) 
                {
                    if (deliveryQuantityNode.Attributes["id"] != null) 
                    {
                        deliveryQuantityIDRef_ = deliveryQuantityNode.Attributes["id"].Value;
                        CommodityPhysicalQuantity ob = new CommodityPhysicalQuantity(deliveryQuantityNode);
                        IDManager.SetID(deliveryQuantityIDRef_, ob);
                    }
                    else if (deliveryQuantityNode.Attributes["href"] != null)
                    {
                        deliveryQuantityIDRef_ = deliveryQuantityNode.Attributes["href"].Value;
                    }
                    else
                    {
                        deliveryQuantity_ = new CommodityPhysicalQuantity(deliveryQuantityNode);
                    }
                }
                else
                {
                    deliveryQuantity_ = new CommodityPhysicalQuantity(deliveryQuantityNode);
                }
            }
            
        
        }
        
    
        #region DeliveryPeriods_
        private CommodityDeliveryPeriods deliveryPeriods_;
        public CommodityDeliveryPeriods DeliveryPeriods_
        {
            get
            {
                if (this.deliveryPeriods_ != null)
                {
                    return this.deliveryPeriods_; 
                }
                else if (this.deliveryPeriodsIDRef_ != null)
                {
                    deliveryPeriods_ = IDManager.getID(deliveryPeriodsIDRef_) as CommodityDeliveryPeriods;
                    return this.deliveryPeriods_; 
                }
                else
                {
                      return this.deliveryPeriods_; 
                }
            }
            set
            {
                if (this.deliveryPeriods_ != value)
                {
                    this.deliveryPeriods_ = value;
                }
            }
        }
        #endregion
        
        public string deliveryPeriodsIDRef_ { get; set; }
        #region Coal_
        private CoalProduct coal_;
        public CoalProduct Coal_
        {
            get
            {
                if (this.coal_ != null)
                {
                    return this.coal_; 
                }
                else if (this.coalIDRef_ != null)
                {
                    coal_ = IDManager.getID(coalIDRef_) as CoalProduct;
                    return this.coal_; 
                }
                else
                {
                      return this.coal_; 
                }
            }
            set
            {
                if (this.coal_ != value)
                {
                    this.coal_ = value;
                }
            }
        }
        #endregion
        
        public string coalIDRef_ { get; set; }
        #region DeliveryConditions_
        private CoalDelivery deliveryConditions_;
        public CoalDelivery DeliveryConditions_
        {
            get
            {
                if (this.deliveryConditions_ != null)
                {
                    return this.deliveryConditions_; 
                }
                else if (this.deliveryConditionsIDRef_ != null)
                {
                    deliveryConditions_ = IDManager.getID(deliveryConditionsIDRef_) as CoalDelivery;
                    return this.deliveryConditions_; 
                }
                else
                {
                      return this.deliveryConditions_; 
                }
            }
            set
            {
                if (this.deliveryConditions_ != value)
                {
                    this.deliveryConditions_ = value;
                }
            }
        }
        #endregion
        
        public string deliveryConditionsIDRef_ { get; set; }
        #region DeliveryQuantity_
        private CommodityPhysicalQuantity deliveryQuantity_;
        public CommodityPhysicalQuantity DeliveryQuantity_
        {
            get
            {
                if (this.deliveryQuantity_ != null)
                {
                    return this.deliveryQuantity_; 
                }
                else if (this.deliveryQuantityIDRef_ != null)
                {
                    deliveryQuantity_ = IDManager.getID(deliveryQuantityIDRef_) as CommodityPhysicalQuantity;
                    return this.deliveryQuantity_; 
                }
                else
                {
                      return this.deliveryQuantity_; 
                }
            }
            set
            {
                if (this.deliveryQuantity_ != value)
                {
                    this.deliveryQuantity_ = value;
                }
            }
        }
        #endregion
        
        public string deliveryQuantityIDRef_ { get; set; }
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-com-5-4/CoalPhysicalLeg.cpp
// CoalPhysicalLeg.cpp 
#include "CoalPhysicalLeg.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

CoalPhysicalLeg::CoalPhysicalLeg(TiXmlNode* xmlNode)
: PhysicalSwapLeg(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //deliveryPeriodsNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* deliveryPeriodsNode = xmlNode->FirstChildElement("deliveryPeriods");

   if(deliveryPeriodsNode){deliveryPeriodsIsNull_ = false;}
   else{deliveryPeriodsIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- deliveryPeriodsNode , address : " << deliveryPeriodsNode << std::endl;
   #endif
   if(deliveryPeriodsNode)
   {
      if(deliveryPeriodsNode->Attribute("href") || deliveryPeriodsNode->Attribute("id"))
      {
          if(deliveryPeriodsNode->Attribute("id"))
          {
             deliveryPeriodsIDRef_ = deliveryPeriodsNode->Attribute("id");
             deliveryPeriods_ = boost::shared_ptr<CommodityDeliveryPeriods>(new CommodityDeliveryPeriods(deliveryPeriodsNode));
             deliveryPeriods_->setID(deliveryPeriodsIDRef_);
             IDManager::instance().setID(deliveryPeriodsIDRef_,deliveryPeriods_);
          }
          else if(deliveryPeriodsNode->Attribute("href")) { deliveryPeriodsIDRef_ = deliveryPeriodsNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { deliveryPeriods_ = boost::shared_ptr<CommodityDeliveryPeriods>(new CommodityDeliveryPeriods(deliveryPeriodsNode));}
   }

   //coalNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* coalNode = xmlNode->FirstChildElement("coal");

   if(coalNode){coalIsNull_ = false;}
   else{coalIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- coalNode , address : " << coalNode << std::endl;
   #endif
   if(coalNode)
   {
      if(coalNode->Attribute("href") || coalNode->Attribute("id"))
      {
          if(coalNode->Attribute("id"))
          {
             coalIDRef_ = coalNode->Attribute("id");
             coal_ = boost::shared_ptr<CoalProduct>(new CoalProduct(coalNode));
             coal_->setID(coalIDRef_);
             IDManager::instance().setID(coalIDRef_,coal_);
          }
          else if(coalNode->Attribute("href")) { coalIDRef_ = coalNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { coal_ = boost::shared_ptr<CoalProduct>(new CoalProduct(coalNode));}
   }

   //deliveryConditionsNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* deliveryConditionsNode = xmlNode->FirstChildElement("deliveryConditions");

   if(deliveryConditionsNode){deliveryConditionsIsNull_ = false;}
   else{deliveryConditionsIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- deliveryConditionsNode , address : " << deliveryConditionsNode << std::endl;
   #endif
   if(deliveryConditionsNode)
   {
      if(deliveryConditionsNode->Attribute("href") || deliveryConditionsNode->Attribute("id"))
      {
          if(deliveryConditionsNode->Attribute("id"))
          {
             deliveryConditionsIDRef_ = deliveryConditionsNode->Attribute("id");
             deliveryConditions_ = boost::shared_ptr<CoalDelivery>(new CoalDelivery(deliveryConditionsNode));
             deliveryConditions_->setID(deliveryConditionsIDRef_);
             IDManager::instance().setID(deliveryConditionsIDRef_,deliveryConditions_);
          }
          else if(deliveryConditionsNode->Attribute("href")) { deliveryConditionsIDRef_ = deliveryConditionsNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { deliveryConditions_ = boost::shared_ptr<CoalDelivery>(new CoalDelivery(deliveryConditionsNode));}
   }

   //deliveryQuantityNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* deliveryQuantityNode = xmlNode->FirstChildElement("deliveryQuantity");

   if(deliveryQuantityNode){deliveryQuantityIsNull_ = false;}
   else{deliveryQuantityIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- deliveryQuantityNode , address : " << deliveryQuantityNode << std::endl;
   #endif
   if(deliveryQuantityNode)
   {
      if(deliveryQuantityNode->Attribute("href") || deliveryQuantityNode->Attribute("id"))
      {
          if(deliveryQuantityNode->Attribute("id"))
          {
             deliveryQuantityIDRef_ = deliveryQuantityNode->Attribute("id");
             deliveryQuantity_ = boost::shared_ptr<CommodityPhysicalQuantity>(new CommodityPhysicalQuantity(deliveryQuantityNode));
             deliveryQuantity_->setID(deliveryQuantityIDRef_);
             IDManager::instance().setID(deliveryQuantityIDRef_,deliveryQuantity_);
          }
          else if(deliveryQuantityNode->Attribute("href")) { deliveryQuantityIDRef_ = deliveryQuantityNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { deliveryQuantity_ = boost::shared_ptr<CommodityPhysicalQuantity>(new CommodityPhysicalQuantity(deliveryQuantityNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<CommodityDeliveryPeriods> CoalPhysicalLeg::getDeliveryPeriods()
{
   if(!this->deliveryPeriodsIsNull_){
        if(deliveryPeriodsIDRef_ != ""){
             return boost::shared_static_cast<CommodityDeliveryPeriods>(IDManager::instance().getID(deliveryPeriodsIDRef_));
        }else{
             return this->deliveryPeriods_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<CommodityDeliveryPeriods>();
   }
}
boost::shared_ptr<CoalProduct> CoalPhysicalLeg::getCoal()
{
   if(!this->coalIsNull_){
        if(coalIDRef_ != ""){
             return boost::shared_static_cast<CoalProduct>(IDManager::instance().getID(coalIDRef_));
        }else{
             return this->coal_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<CoalProduct>();
   }
}
boost::shared_ptr<CoalDelivery> CoalPhysicalLeg::getDeliveryConditions()
{
   if(!this->deliveryConditionsIsNull_){
        if(deliveryConditionsIDRef_ != ""){
             return boost::shared_static_cast<CoalDelivery>(IDManager::instance().getID(deliveryConditionsIDRef_));
        }else{
             return this->deliveryConditions_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<CoalDelivery>();
   }
}
boost::shared_ptr<CommodityPhysicalQuantity> CoalPhysicalLeg::getDeliveryQuantity()
{
   if(!this->deliveryQuantityIsNull_){
        if(deliveryQuantityIDRef_ != ""){
             return boost::shared_static_cast<CommodityPhysicalQuantity>(IDManager::instance().getID(deliveryQuantityIDRef_));
        }else{
             return this->deliveryQuantity_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<CommodityPhysicalQuantity>();
   }
}
}

